src/HeaderCell.tsx

```diff
@@ -218,10 +223,12 @@ export default function HeaderCell<R, SR>({
 
   function onDrop(event: React.DragEvent<HTMLDivElement>) {
     setIsOver(false);
-    const sourceKey = event.dataTransfer.getData('text/plain');
-    if (sourceKey !== column.key) {
-      event.preventDefault();
-      onColumnsReorder?.(sourceKey, column.key);
+    if (event.dataTransfer.types.includes(dragDropKey)) {
+      const sourceKey = event.dataTransfer.getData(dragDropKey);
```

Comment:
I can probably add an undefined check instead but need to ignore the types
![image](https://github.com/adazzle/react-data-grid/assets/1452717/306304c7-8c16-4f60-a1ad-a4573951124f)